### PR TITLE
IO Optimizations

### PR DESCRIPTION
**Pull request by DJShepherd**
* Reduced memory copies in NAEUSB and serial interfaces and refactored API
* Refactored OpenADCInterface sendMessage API
* Added optimized memory utility functions
* Renamed CW/bytearray to CWByteArray to prevent type confusions with builtin functions

Passed all tests/test_api.py and ran the clock glitch demo on CWLITE.

### CODE REVIEW

**Comment by alex-dewar** [comment]
Hi,

Thanks for making the changes! I'll take a look through the code and provide feedback if necessary. My initial run through some of the tutorials with the Lite looks great.

File: software/chipwhisperer/capture/acq_patterns/basic.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from ...common.utils import util
 from ._base import AcqKeyTextPattern_Base
 from ...common.utils.util import camel_case_deprecated
-bytearray = util.bytearray # type: ignore
+bytearray = util.CWByteArray # type: ignore
```
Comment (alex-dewar) [inline software/chipwhisperer/capture/acq_patterns/basic.py:30]: Lines 190 and 209 are using `util.bytearray()`, which needs to be changed to `bytearray()`

Comment (DJShepherd) [reply software/chipwhisperer/capture/acq_patterns/basic.py:30]: Fixed! Also grepped the project for other similar instances and fixed those, as well.

**Comment by alex-dewar** [comment]
Code looks good and seems to run fine.

**Pull request merged**